packages/x-charts-pro/src/internals/plugins/useChartProZoom/useChartProZoom.types.ts

```diff
@@ -11,17 +11,17 @@ export interface UseChartProZoomParameters {
    * The list of zoom data related to each axis.
    * Used to initialize the zoom in a specific configuration without controlling it.
    */
-  initialZoom?: ZoomData[];
+  initialZoom?: readonly ZoomData[];
   /**
    * Callback fired when the zoom has changed.
    *
    * @param {ZoomData[]} zoomData Updated zoom data.
    */
-  onZoomChange?: (zoomData: ZoomData[] | ((zoomData: ZoomData[]) => ZoomData[])) => void;
+  onZoomChange?: (zoomData: ZoomData[]) => void;
```

Comment:
Shouldn't `zoomData` here also be `readonly`? We probably don't want users `push`ing to this array, right?

Reply:
Technically, yes, but I didn't want to introduce a breaking change with this 😩

Reply:
Makes sense 👍 

Isn't now the best time to do it? Or is the performance hit preferable to causing a breaking change? Still trying to understand where we should place the line.

Reply:
Right now there is no performance hit, I've removed the destructuring changes by making the internal state also immutable.

Reply:
Sorry, I should have linked the line I was referring to. 

I was referring to line 106 on `useChartProZoom`. We're now creating a new array where we weren't before.

Reply:
Ah, true, but that specific case shouldn't really affect us much. `zoomData` is not meant to be a super long array.